src/modules/sub-items/components/table-view/table.view.component.js

```diff
@@ -231,12 +232,18 @@ export default class TableViewComponent extends Component {
         const { columnsVisibility } = this.state;
         const { selectedLocationsIds } = this.props;
         const anyLocationSelected = !!selectedLocationsIds.size;
+        const allLocationsSelected = selectedLocationsIds.size === items.length;
+        const isCheckboxIndeterminate = anyLocationSelected && !allLocationsSelected;
 
         return (
             <thead className={headClass}>
                 <tr className="c-table-view__row">
                     <th className={`${TABLE_HEAD_CLASS} ${cellClass}--checkbox`}>
-                        <input type="checkbox" checked={anyLocationSelected} onChange={this.selectAll} />
+                        <ThreeStateCheckboxComponent
+                            indeterminate={isCheckboxIndeterminate}
+                            checked={anyLocationSelected}
+                            onClick={this.selectAll}
```

Comment:
Note: we need `onClick`, because MS Edge does not trigger `onChange` when checkbox has _indeterminate state_. (ref: https://stackoverflow.com/a/33529024/5766602)

Reply:
I think this is a good candidate for a small comment in the code. This is unnatural to have `onClick` on the checkbox.

Reply:
Comment added :)